### ISSUE TITLE
segment function of Request service returns urlencoded string

### ISSUE THREAD

**Issue by TheAliAbbasi** [laravel/framework#15492]
- Laravel Version: 5.2.43
- PHP Version: 7.0.10
- Database Driver & Version: Mysql 5.5, PDO

### Description:
I tried to use UTF-8 in my URI and in some sceniaros I need to fetch that data using Request::segment(2) and this function returns urldecoded version and I have to decode it wherever I use it

**Comment by themsaid** [comment]
@TheAliAbbasi your issue is missing the "Steps to reproduce" section, please provide an example on your issue.

**Comment by TheAliAbbasi** [comment]
Reproduce steps:
Make a reoute like following
```php Route::get('{segment}', function(){
        dd(\Request::segment(1));
}); ```

Then call it with utf-8 characters such as سلام
You'll the urlencoded version of thay word

**Comment by srmklive** [comment]
@TheAliAbbasi Do you have `php_intl` extension enabled?

**Comment by TheAliAbbasi** [comment]
@srmklive nope, but I installed it and no differences. I can ```php urldecode() ``` the segment on my own, and other functions such as input() or only() functions or in router binder. it's just this function is excluded

**Comment by jarnovanleeuwen** [comment]
Symfony's `getPathInfo()` returns deliberately a urlencoded path. Laravel provides access to a urldecoded path through `Request::decodedPath()`, but not to segments. I think you could easily write a wrapper for the segment function that returns urldecoded segments.

**Comment by TheAliAbbasi** [comment]
could you fix that in 5.2 as well?!

**Comment by themsaid** [comment]
5.2 is closed :)
